### ISSUE TITLE
Refersh for 2d

### ISSUE THREAD

**Issue by mkmdivy** [vasturiano/react-force-graph#315]
Hi Thank you for the great package! 

I was trying to .refresh() for the graph with fgRef.current and kept get the error and I realized that it is supported for 3d ar vr but not for 2d. Is it planned to added for 2d too?

**Comment by vasturiano** [comment]
@mkmdivy you should normally not ever need to do `refresh` on the module. Specially for the 2d mode which doesn't have any type of objects cached in internal state (everything is drawn based on the stateless render functions). Can you describe more in detail what you are trying to achieve, maybe there's a better way.

**Comment by mkmdivy** [comment]
@vasturiano Hi, thank you for your reply! Maybe my example can be helpful to describe my situation. I am changing dynamically the length(distance) of the edge. 

Here is my example, 
https://africapolis.org/network/ you can click network.

Here, the edge is travel time between nodes, and I change the length(distance) of the edge dynamically with 0,60,300,1000 minutes. Initially it is set as 1000, but when you click 0 and then set 1000 again, the shape is a bit different. (Actually the graph was keep getting bigger but after I set iterations as 200 it is much more stable) I also set the x and y manually as initial position but then I think velocity is all different it behaves different. 

So, in short, what I want is, I want always same result when I click 1000 minutes, regardless I chose different minutes. I think it is affected by previous position.

**Comment by dvargas92495** [comment]
>  you should normally not ever need to do refresh on the module. Specially for the 2d mode which doesn't have any type of objects cached in internal state (everything is drawn based on the stateless render functions)

Is this because the library is mutating the input data?

I have a use case where on hover of one node, I want to change the opacity of other nodes. but because I pass in the initial node data as a filter of data from my db, rerenders always resets the position of the nodes in the canvas, causing a jarring interaction. In this case, I would love a refresh method so that ForceGraph2D could maintain the positions and other stylings, without mutating my input data

**Comment by vasturiano** [comment]
@dvargas92495 yes the force simulation stores positional information in the node objects. You can read more about it in the `d3-force` repo: https://github.com/d3/d3-force#simulation_nodes

Furthermore the comparison of whether a node is new or existing is done by object reference, therefore if you are injecting data from a backend (including potentially the same nodes), you should have a reconciliation function which reuses the existing objects (that you can extract via the getter `.graphData()`) by means of a node ID or similar, and reuse the objects.

Similarly, the same thing happens with the conversion of the link's `source` and `target` attributes: https://github.com/d3/d3-force#link_links
